configuration/configurator/ConfigSchemaHelper.cpp

```diff
@@ -0,0 +1,889 @@
+                static StringBuffer strDoc;
+                strDoc.clear(); // needed when printing more than 1 component
+                pSchema->getDocumentation(strDoc);
+
+                return strDoc.str();
+             }
+        }
+    }
+
+    return NULL;
+}
+
+
+const char* CConfigSchemaHelper::printDojoJS(const char* comp)
+{
+    assert(comp != NULL && *comp != 0);
+    assert(m_pSchemaMapManager != NULL);
+
+    if (comp == NULL || *comp == 0)
+    {
+        return NULL;
+    }
+
+    CSchema* pSchema = NULL;
+
+    LOOP_THRU_BUILD_SET_MANAGER_BUILD_SET
+    {
+        const char *pSchemaName = CBuildSetManager::getInstance()->getBuildSetComponentFileName(idx);
+
+        if (pSchemaName != NULL && strcmp(comp, pSchemaName) == 0)
+        {
+             pSchema = m_pSchemaMapManager->getSchemaForXSD(pSchemaName);
+
+             assert(pSchema != NULL);
+
+             if (pSchema != NULL)
+             {
+                static StringBuffer strDoc;
+                pSchema->getDojoJS(strDoc);
+
+                return strDoc.str();
+             }
+        }
+    }
+
+    return NULL;
+}
+
+const char* CConfigSchemaHelper::printQML(const char* comp, int nIdx) const
+{
+    //assert(comp != NULL && *comp != 0);
+    if (! (comp != NULL && *comp != 0) )
+    {
+        DBGLOG("no component selected for QML, index = %d", nIdx);
+        return NULL;
+    }
+    assert(m_pSchemaMapManager != NULL);
+
+    static StringBuffer strQML;
+
+    strQML.clear();
+    resetTables();
+
+    if (comp == NULL || *comp == 0)
+    {
+        return NULL;
+    }
+
+    CSchema* pSchema = NULL;
+
+    LOOP_THRU_BUILD_SET_MANAGER_BUILD_SET
+    {
+        const char *pSchemaName = CBuildSetManager::getInstance()->getBuildSetComponentFileName(idx);
+
+        if (pSchemaName != NULL && strcmp(comp, pSchemaName) == 0)
+        {
+             pSchema = m_pSchemaMapManager->getSchemaForXSD(pSchemaName);
+
+             assert(pSchema != NULL);
+
+             if (pSchema != NULL)
+             {
+                 pSchema->getQML3(strQML, nIdx);
+                 return strQML.str();
+             }
+        }
+    }
+
+    return NULL;
+}
+
+void CConfigSchemaHelper::printDump(const char* comp) const
+{
+    assert(comp != NULL && *comp != 0);
+    assert(m_pSchemaMapManager != NULL);
+
+    if (comp == NULL || *comp == 0)
+    {
+        return;
+    }
+
+    CSchema* pSchema = NULL;
+
+    LOOP_THRU_BUILD_SET_MANAGER_BUILD_SET
+    {
+        const char *pSchemaName = CBuildSetManager::getInstance()->getBuildSetComponentFileName(idx);
+
+        if (pSchemaName != NULL && strcmp(comp, pSchemaName) == 0)
+        {
+             pSchema = m_pSchemaMapManager->getSchemaForXSD(pSchemaName);
+
+             assert(pSchema != NULL);
+
+             if (pSchema != NULL)
+             {
+                pSchema->dump(std::cout);
+            }
+        }
+    }
+}
+
+void CConfigSchemaHelper::dumpStdOut() const
+{
+    assert("NOT IMPLEMENTED");
```

Comment:
Is it something that needs to be implemented some time? In which case the UNIMPLEMENTED macro would be appropriate. Or is it something that should never be executed (e.g. overridden in derived classes)? In which case throwUnexpected() would be appropriate.